pypi_compatible_build.py

```diff
@@ -0,0 +1,30 @@
+"""
+Avoid git URLs breaking PyPI uploads.
+
+Similar problem to:
+https://stackoverflow.com/questions/54887301/how-can-i-use-git-repos-as-dependencies-for-my-pypi-package
+"""
+
+from io import StringIO
+from typing import TextIO
+
+import setuptools
+from packaging.metadata import Metadata
+from setuptools._core_metadata import _write_requirements  # type: ignore[import-untyped]
```

Comment:
Why do you have this if you have `types-setuptools`?

Reply:
It looks like the stubs don't cover `_core_metadata` https://github.com/python/typeshed/tree/main/stubs/setuptools/setuptools